internal/provider/newrelic/provider_integration_test.go

```diff
@@ -168,12 +177,155 @@ func Test_Getting_external_metric_through_proxy(t *testing.T) {
 	})
 }
 
-func newrelicProviderWithMetric(t *testing.T, metric newrelic.Metric) provider.ExternalMetricsProvider {
-	t.Helper()
+// This test uses the `Region: Local` which configures the newrelic go client to use the `localhost:3000` endpoint
+// (https://github.com/newrelic/newrelic-client-go/blob/main/pkg/region/region_constants.go).
+//
+//nolint:funlen,tparallel,paralleltest // no parallel since port 3000 is used by all test.
+func Test_does_not_hide_backend_errors(t *testing.T) {
+	t.Parallel()
 
-	testEnv := testutil.TestEnv{}
+	ctx := testutil.ContextWithDeadline(t)
+
+	testEnv := testutil.TestEnv{
+		StartKubernetes: true,
+		Region:          "Local",
+	}
 	testEnv.Generate(t)
 
+	url := fmt.Sprintf("%s/apis/external.metrics.k8s.io/v1beta1/namespaces/*/test_metric", testEnv.BaseURL)
+
+	p := newrelicProviderWithMetric(t, newrelic.Metric{
+		Query: testIntegrationQuery,
+	}, &testEnv)
+
+	runAdapter(t, &testEnv, p)
+
+	t.Run("when_newrelic_backend_response_error_is", func(t *testing.T) {
+		cases := map[string]func(w http.ResponseWriter, r *http.Request){
+			"200_with_empty_data": func(w http.ResponseWriter, _ *http.Request) {
+				w.WriteHeader(http.StatusOK)
+			},
+			"400": func(w http.ResponseWriter, _ *http.Request) {
+				w.WriteHeader(http.StatusBadRequest)
+			},
+			"401": func(w http.ResponseWriter, _ *http.Request) {
+				w.WriteHeader(http.StatusUnauthorized)
+			},
+			"403": func(w http.ResponseWriter, _ *http.Request) {
+				w.WriteHeader(http.StatusForbidden)
+			},
+			"500": func(w http.ResponseWriter, _ *http.Request) {
+				w.WriteHeader(http.StatusInternalServerError)
+			},
+		}
+
+		for testCaseName, handlerFunc := range cases {
+			handlerFunc := handlerFunc
+			t.Run(testCaseName, func(t *testing.T) {
+				runHTTPTestServer(t, 3000, handlerFunc)
+
+				m := provider.ExternalMetricInfo{Metric: testMetricName}
+
+				if _, err := p.GetExternalMetric(ctx, "", nil, m); err == nil {
+					t.Errorf("Unexpected error = %v", err)
+				}
+
+				testutil.RetryGetRequestAndCheckStatus(ctx, t, testEnv.HTTPClient, url, func(statusCode int) bool {
+					return statusCode == http.StatusOK
```

Comment:
nit: if any issue running this test this will blown up after default test timeout if I'm not mistaken (5m) and might be tricky to get from the logs. It might be good to restrict the retries to a timeout or something like that.

Reply:
This context is set to max 10 seconds as stated in the `testutil.ContextWithDeadline`  func.